pkg/contexts/ocm/accessmethods/mvn/method.go

```diff
@@ -0,0 +1,408 @@
+}
+
+func filesAndHashes(fileList []string) map[string]crypto.Hash {
+	// Sort the list of files, to ensure always the same results for e.g. identical tar.gz files.
+	sort.Strings(fileList)
+
+	// Which hash files are available?
+	result := make(map[string]crypto.Hash)
+	for _, file := range fileList {
+		if IsResource(file) {
+			result[file] = bestAvailableHash(fileList, file)
+			log.Debug("found", "file", file)
+		}
+	}
+	return result
+}
+
+// bestAvailableHash returns the best available hash for the given file.
+// It first checks for SHA-512, then SHA-256, SHA-1, and finally MD5. If nothing is found, it returns 0.
+func bestAvailableHash(list []string, filename string) crypto.Hash {
+	hashes := [5]crypto.Hash{crypto.SHA512, crypto.SHA256, crypto.SHA1, crypto.MD5}
+	for _, hash := range hashes {
+		if slices.Contains(list, filename+hashUrlExt(hash)) {
+			return hash
+		}
+	}
+	return 0
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+// getStringData reads all data from the given URL and returns it as a string.
+func getStringData(ctx accspeccpi.Context, url string, fs vfs.FileSystem) (string, error) {
+	r, err := getReader(ctx, url, fs)
+	if err != nil {
+		return "", err
+	}
+	b, err := io.ReadAll(r)
+	if err != nil {
+		return "", err
+	}
+	return string(b), nil
+}
+
+// hashUrlExt returns the 'maven' hash extension for the given hash.
+// Maven usually uses sha1, sha256, sha512, md5 instead of SHA-1, SHA-256, SHA-512, MD5.
+func hashUrlExt(h crypto.Hash) string {
+	return "." + strings.ReplaceAll(strings.ToLower(h.String()), "-", "")
+}
+
+func newMethod(c accspeccpi.ComponentVersionAccess, a *AccessSpec) (accspeccpi.AccessMethodImpl, error) {
+	factory := func() (blobaccess.BlobAccess, error) {
+		meta, err := a.GetPackageMeta(c.GetContext())
+		if err != nil {
+			return nil, err
+		}
+
+		reader := func() (io.ReadCloser, error) {
+			return getReader(c.GetContext(), meta.Bin, vfsattr.Get(c.GetContext()))
+		}
+		if meta.Hash != "" {
+			getreader := reader
+			reader = func() (io.ReadCloser, error) {
+				readCloser, err := getreader()
+				if err != nil {
+					return nil, err
+				}
+				return iotools.VerifyingReaderWithHash(readCloser, meta.HashType, meta.Hash), nil
+			}
+		}
+		acc := blobaccess.DataAccessForReaderFunction(reader, meta.Bin)
+		return accessobj.CachedBlobAccessForWriter(c.GetContext(), a.MimeType(), accessio.NewDataAccessWriter(acc)), nil
+	}
+	// FIXME add Digest!
+	return accspeccpi.NewDefaultMethodImpl(c, a, "", a.MimeType(), factory), nil
+}
+
+func getReader(ctx accspeccpi.Context, url string, fs vfs.FileSystem) (io.ReadCloser, error) {
+	if strings.HasPrefix(url, "file://") {
+		path := url[7:]
+		return fs.OpenFile(path, vfs.O_RDONLY, 0o600)
+	}
+
+	req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+	identity.BasicAuth(req, ctx, url, "")
+	httpClient := &http.Client{}
```

Comment:
Normally, we don't use plain http client. Set some defaults like this:

At least set `Timeout: time.Second * 10`.

Reply:
together with https://github.com/open-component-model/ocm/issues/755 we might think of an OCM Library specific http-client, what do you think?

Reply:
will be handled in follow up task:
https://github.com/open-component-model/ocm/issues/755#issuecomment-2115003052